server/src/controllers/attainment.ts

```diff
@@ -313,3 +298,37 @@ export async function getAttainment(req: Request, res: Response): Promise<void>
     data: attainment,
   });
 }
+
+export async function getAllAttainments(req: Request, res: Response): Promise<void> {
+  const courseId: number = Number(req.params.courseId);
+  const instanceId: number = Number(req.params.instanceId);
+
+  await idSchema.validate({ id: courseId }, { abortEarly: false });
+  await idSchema.validate({ id: instanceId }, { abortEarly: false });
+  await treeSchema.validate(req.query, { abortEarly: false });
+
+  // Assert string type, as the query is validated above
+  const tree: string | undefined = req.query.tree as string | undefined;
+
+  const allAttainmentData: Array<AttainmentData> = await findAllAttainmentsForInstance(
+    courseId,
+    instanceId
+  );
+
+  const rootAttainments: Array<AttainmentData> = allAttainmentData.filter(
```

Comment:
We've been working with the idea that there should only ever be one root attainment which determines the final grade for a course. This may end up being subject to change, but right now some other code treats it as an error state that there is more than one root attainment, so for now this endpoint should also assume that only one root attainment exists.

Reply:
Yes, I imagined this endpoint would serve the purpose of not needing to find the root attainment's ID to get all attainments of a course instance. This could also check that only one attainment with no parent exists on the course instance, and if not, return the IDs of all the attainments with no parent in an error message.